test/jobs/TestJobsFeature.cpp

```diff
@@ -710,3 +730,39 @@ TEST_F(TestJobsFeature, InvalidJobDocument)
     jobsMock->init(std::shared_ptr<Mqtt::MqttConnection>(), notifier, config);
     jobsMock->invokeRunJobs();
 }
+
+    TEST_F(TestJobsFeaturePrivate, CompareJobDocuments)
+{
+    // Test case 1: Identical documents
+    Aws::Crt::JsonObject doc1("{\"key\": \"value\"}");
+    Aws::Crt::JsonObject doc2("{\"key\": \"value\"}");
+    EXPECT_TRUE(compareJobDocuments(doc1, doc2));
+
+    // Test case 2: Different documents
+    Aws::Crt::JsonObject doc3("{\"key\": \"different_value\"}");
+    EXPECT_FALSE(compareJobDocuments(doc1, doc3));
+
+    // Test case 3: Documents with pre-signed URLs
+    Aws::Crt::JsonObject doc4("{\"url\": \"https://bucket.s3.amazonaws.com/file?AWSAccessKeyId=AKIAIOSFODNN7EXAMPLE&Expires=1234567890&Signature=XXXXXXXXXXXXXXXXXXXXXXXXXXX\"}");
+    Aws::Crt::JsonObject doc5("{\"url\": \"https://bucket.s3.amazonaws.com/file?AWSAccessKeyId=AKIAIOSFODNN7EXAMPLE&Expires=9876543210&Signature=YYYYYYYYYYYYYYYYYYYYYYYYYYY\"}");
+    EXPECT_TRUE(compareJobDocuments(doc4, doc5));
+
+    // Test case 4: Documents with multiple pre-signed URLs
+    Aws::Crt::JsonObject doc6("{\"url1\": \"https://bucket1.s3.amazonaws.com/file1?AWSAccessKeyId=AKIAIOSFODNN7EXAMPLE&Expires=1234567890&Signature=XXXXXXXXXXXXXXXXXXXXXXXXXXX\", \"url2\": \"https://bucket2.s3.amazonaws.com/file2?AWSAccessKeyId=AKIAIOSFODNN7EXAMPLE&Expires=1234567890&Signature=ZZZZZZZZZZZZZZZZZZZZZZZZZZZ\"}");
+    Aws::Crt::JsonObject doc7("{\"url1\": \"https://bucket1.s3.amazonaws.com/file1?AWSAccessKeyId=AKIAIOSFODNN7EXAMPLE&Expires=9876543210&Signature=YYYYYYYYYYYYYYYYYYYYYYYYYYY\", \"url2\": \"https://bucket2.s3.amazonaws.com/file2?AWSAccessKeyId=AKIAIOSFODNN7EXAMPLE&Expires=9876543210&Signature=WWWWWWWWWWWWWWWWWWWWWWWWWWW\"}");
+    EXPECT_TRUE(compareJobDocuments(doc6, doc7));
+
+    // Test case 6: Documents with multiple pre-signed URLs in one doc and single url in the other
```

Comment:
Are we testing the scenario where there are multiple pre-signed URLs in both documents (which may or may not be the same URLs in both documents), but the order is different?

For example:

```
    Aws::Crt::JsonObject doc6("{\"url1\": \"https://bucket1.s3.amazonaws.com/file1?AWSAccessKeyId=AKIAIOSFODNN7EXAMPLE&Expires=1234567890&Signature=XXXXXXXXXXXXXXXXXXXXXXXXXXX\", \"url2\": \"https://bucket2.s3.amazonaws.com/file2?AWSAccessKeyId=AKIAIOSFODNN7EXAMPLE&Expires=1234567890&Signature=ZZZZZZZZZZZZZZZZZZZZZZZZZZZ\"}");
    Aws::Crt::JsonObject doc7("{\"url1\": \"https://bucket2.s3.amazonaws.com/file2?AWSAccessKeyId=AKIAIOSFODNN7EXAMPLE&Expires=9876543210&Signature=WWWWWWWWWWWWWWWWWWWWWWWWWWW\", \"url2\": \"https://bucket1.s3.amazonaws.com/file1?AWSAccessKeyId=AKIAIOSFODNN7EXAMPLE&Expires=9876543210&Signature=YYYYYYYYYYYYYYYYYYYYYYYYYYY\"}");
```

Reply:
Considering that the job documents are sent by IOT core due to the subscription mechanisms, it is not possible in theory for the order of the 2 job documents to differ which has been confirmed via manual testing as well. Still, if they differ only in the order of the URLs, it is most likely cause of an actual separate job sent and hence they should be inferred as a different jobs altogether.